ui/app/models/oidc/key.js

```diff
@@ -1,27 +1,59 @@
 import Model, { attr } from '@ember-data/model';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
+const validations = {
+  name: [
+    { type: 'presence', message: 'Name is required.' },
+    {
+      type: 'containsWhiteSpace',
+      message: 'Name cannot contain whitespace.',
+    },
+  ],
+};
+
+@withModelValidations(validations)
 export default class OidcKeyModel extends Model {
-  @attr('string') name;
+  @attr('string', { editDisabled: true }) name;
 
   @attr('string', {
     defaultValue: 'RS256',
     possibleValues: ['RS256', 'RS384', 'RS512', 'ES256', 'ES384', 'ES512', 'EdDSA'],
   })
   algorithm;
 
-  @attr({ editType: 'ttl', helperTextDisabled: 'Vault will use the default lease duration' }) rotationPeriod;
+  @attr({
+    editType: 'ttl',
+    defaultValue: '24h',
+    helperTextDisabled: 'Vault will use the default lease duration',
```

Comment:
NIT: Looking at the other models that use helperTextDisabled they include a period at the end of the sentence.